pkg/models/datasource.go

```diff
@@ -202,9 +202,10 @@ type DeleteDataSourceByNameCommand struct {
 // QUERIES
 
 type GetDataSourcesQuery struct {
-	OrgId  int64
-	User   *SignedInUser
-	Result []*DataSource
+	OrgId           int64
+	DataSourceLimit *int
```

Comment:
I don't think this should be a pointer, instead we should use 0 to indicate that there's no limit. It's more idiomatic Go, and less error prone.

Reply:
I have suggested to use a pointer to imply that there is not limit set intentionally.
But I'm open to discuss it.

Reply:
@papagian AFAICT it changes behaviour though, so calls that would previously be limited to 5000 are unlimited.

I think it shouldn't be a pointer, but probably 5000 by default if `DataSourceLimit` is 0 (in case callers doesn't set it). Maybe when reading the configuration, `DataSourceLimit` should be set to -1 if the user passes zero to indicate unlimited? WDYT?

Reply:
This is discussed in a previous comment thread 🙂

Reply:
I wrote some additional tests and I've verified that if the [limit](https://godoc.org/github.com/go-xorm/xorm#Engine.Limit) is zero or negative it will be ignored.
Therefore I agree that it will simplify the code not to be a pointer.
@ying-jeanne can you change it?

Reply:
@papagian @ying-jeanne Can we explicitly not set the limit if it's zero, since [Engine.Limit](https://godoc.org/github.com/go-xorm/xorm#Engine.Limit) doesn't define zero limit as equivalent to unlimited? I.e., we shouldn't depend on undefined behaviour and it would also make the intent obvious to readers of our code.